library/src/main/java/com/klinker/android/send_message/Transaction.java

```diff
@@ -707,13 +713,15 @@ private static void sendMmsThroughSystem(Context context, String subject, List<M
         }
     }
 
-    private static SendReq buildPdu(Context context, String[] recipients, String subject,
+    private static SendReq buildPdu(Context context, String fromAddress, String[] recipients, String subject,
                                     List<MMSPart> parts) {
         final SendReq req = new SendReq();
         // From, per spec
-        final String lineNumber = Utils.getMyPhoneNumber(context);
-        if (!TextUtils.isEmpty(lineNumber)) {
-            req.setFrom(new EncodedStringValue(lineNumber));
+        String phoneNumber = Utils.getMyPhoneNumberFromSubscription(context, settings.getSubscriptionId());
+        if (!TextUtils.isEmpty(phoneNumber)) {
+            req.setFrom(new EncodedStringValue(phoneNumber));
+        } else if (!TextUtils.isEmpty(fromAddress)) {
+            req.setFrom(new EncodedStringValue(fromAddress));
```

Comment:
Same comments as above. Would probably be useful to pull this into a function. You could make the new function within the `SendReq` class, probably.

Reply:
makes sense